src/posix/platform/radio.cpp

```diff
@@ -0,0 +1,451 @@
+
+bool otPlatRadioIsEnabled(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    return sRadioSpinel.IsEnabled();
+}
+
+otError otPlatRadioEnable(otInstance *aInstance)
+{
+    return sRadioSpinel.Enable(aInstance);
+}
+
+otError otPlatRadioDisable(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    return sRadioSpinel.Disable();
+}
+
+otError otPlatRadioSleep(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    return sRadioSpinel.Sleep();
+}
+
+otError otPlatRadioReceive(otInstance *aInstance, uint8_t aChannel)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    return sRadioSpinel.Receive(aChannel);
+}
+
+otError otPlatRadioTransmit(otInstance *aInstance, otRadioFrame *aFrame)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    return sRadioSpinel.Transmit(*aFrame);
+}
+
+otRadioFrame *otPlatRadioGetTransmitBuffer(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    return &sRadioSpinel.GetTransmitFrame();
+}
+
+int8_t otPlatRadioGetRssi(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    return sRadioSpinel.GetRssi();
+}
+
+otRadioCaps otPlatRadioGetCaps(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    return sRadioSpinel.GetRadioCaps();
+}
+
+const char *otPlatRadioGetVersionString(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    return sRadioSpinel.GetVersion();
+}
+
+bool otPlatRadioGetPromiscuous(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    return sRadioSpinel.IsPromiscuous();
+}
+
+void platformRadioUpdateFdSet(fd_set *aReadFdSet, fd_set *aWriteFdSet, int *aMaxFd, struct timeval *aTimeout)
+{
+    sRadioSpinel.GetSpinelInterface().UpdateFdSet(*aReadFdSet, *aWriteFdSet, *aMaxFd, *aTimeout);
+
+    if (sRadioSpinel.IsTransmitting())
+    {
+        uint64_t now          = platformGetTime();
+        uint64_t txRadioEndUs = sRadioSpinel.GetTxRadioEndUs();
+
+        if (now < txRadioEndUs)
+        {
+            uint64_t remain = txRadioEndUs - now;
+
+            if (remain < static_cast<uint64_t>(aTimeout->tv_sec * US_PER_S + aTimeout->tv_usec))
+            {
+                aTimeout->tv_sec  = static_cast<time_t>(remain / US_PER_S);
+                aTimeout->tv_usec = static_cast<suseconds_t>(remain % US_PER_S);
+            }
+        }
+        else
+        {
+            aTimeout->tv_sec  = 0;
+            aTimeout->tv_usec = 0;
+        }
+    }
+
+    if (sRadioSpinel.HasSavedFrame() || sRadioSpinel.IsTransmitDone())
+    {
+        aTimeout->tv_sec  = 0;
+        aTimeout->tv_usec = 0;
+    }
+}
+
+void platformRadioProcess(otInstance *aInstance, const fd_set *aReadFdSet, const fd_set *aWriteFdSet)
+{
+    if (sRadioSpinel.HasSavedFrame())
+    {
+        sRadioSpinel.ProcessFrameQueue();
+    }
+
+    sRadioSpinel.GetSpinelInterface().Process(*aReadFdSet, *aWriteFdSet);
+
+    if (sRadioSpinel.HasSavedFrame())
+    {
+        sRadioSpinel.ProcessFrameQueue();
+    }
+
+    sRadioSpinel.ProcessRadioStateMachine();
```

Comment:
Can we put the code between line 207 and 219 in a function in `sRadioSpinel`? All the functions between line 207 and 219 are accessible in `sRadioSpinel`.

Reply:
Not all platforms provide `fd_set` and `select` so we cannot pass the fd_sets to a member function of RadioSpinel.